libopae/plugins/xfpga/metrics/metrics_utils.c

```diff
@@ -0,0 +1,1231 @@
+			}
+		}
+		return result;
+	}
+
+	result = xfpga_bmcLoadSDRs(_handle, &records, &num_sensors);
+	if (result != FPGA_OK) {
+		FPGA_ERR("Failed to load BMC SDR.");
+		return result;
+	}
+
+	if (_handle->_bmc_metric_value == NULL) {
+		_handle->_bmc_metric_value = malloc(sizeof(struct _fpga_bmc_metric)* num_sensors);
+		if (_handle->_bmc_metric_value == NULL) {
+			FPGA_ERR("Failed to allocate memory");
+			result = FPGA_NO_MEMORY;
+			goto out_destroy;
+		}
+		_handle->num_bmc_metric = num_sensors;
+	}
+
+	result = xfpga_bmcReadSensorValues(_handle, records, &values, &num_values);
+	if (result != FPGA_OK) {
+		FPGA_ERR("Failed to read BMC sensor values.");
+		goto out_destroy;
+	}
+
+	for (x = 0; x < num_sensors; x++) {
+
+		result = xfpga_bmcGetSDRDetails(_handle, values, x, &details);
+		if (result != FPGA_OK) {
+			FPGA_MSG("Failed to get SDR details.");
+		}
+
+		result = xfpga_bmcGetSensorReading(_handle, values, x, &is_valid, &tmp);
+		if (result != FPGA_OK) {
+			FPGA_MSG("Failed to read sensor readings.");
+			continue;
+		}
+
+		if (!is_valid) {
+			continue;
+		}
+
+		snprintf(_handle->_bmc_metric_value[x].metric_name, sizeof(_handle->_bmc_metric_value[x].metric_name), "%s", details.name);
+		_handle->_bmc_metric_value[x].fpga_metric.value.dvalue = tmp;
+
+	}
+
+
+	result = xfpga_bmcDestroySensorValues(_handle, &values);
+	if (result != FPGA_OK) {
+		FPGA_MSG("Failed to Destroy Sensor value.");
+	}
+
+out_destroy:
+	result = xfpga_bmcDestroySDRs(_handle, &records);
+	if (result != FPGA_OK) {
+		FPGA_ERR("Failed to Destroy SDR.");
+		return result;
+	}
+
+	return result;
+}
+fpga_result get_pwr_thermal_value(const char *sysfs_path,
+									uint64_t *value)
+{
+	fpga_result result				= FPGA_OK;
+	char *ptr						= NULL;
+
+	if (sysfs_path == NULL ||
+		value == NULL) {
+		FPGA_ERR("Invlaid Input Paramters");
+		return FPGA_INVALID_PARAM;
+	}
+
+	result = sysfs_read_u64(sysfs_path, value);
+	if (result != FPGA_OK) {
+		FPGA_ERR("Failed to read Metrics values");
+		return result;
+	}
+
+	ptr = strstr(sysfs_path, FPGA_LIMIT);
+	if (ptr)
+		*value = *value / 8;
+
+	ptr = NULL;
+	ptr = strstr(sysfs_path, XEON_LIMIT);
+	if (ptr)
+		*value = *value / 8;
+
+	return result;
+}
+
+fpga_result get_perf_value(const char *group_sysfs,
+							const char *metric_sysfs,
+							uint64_t *value)
+{
+	fpga_result result					= FPGA_OK;
+	char sysfs_path[SYSFS_PATH_MAX]		= { 0 };
+	uint64_t val						= 0;
+
+	if (group_sysfs == NULL ||
+		metric_sysfs == NULL ||
+		value == NULL) {
+		FPGA_ERR("Invlaid Input Paramters");
+		return FPGA_INVALID_PARAM;
+	}
+
+	snprintf_s_ss(sysfs_path, sizeof(sysfs_path), "%s/%s", group_sysfs, PERF_ENABLE);
+
+	result = sysfs_path_is_file(sysfs_path);
+	if (result == FPGA_OK) {
+		result = sysfs_read_u64(sysfs_path, &val);
+		if (result != FPGA_OK) {
+			FPGA_ERR("Failed to read perf fabric enable");
+		}
+
+		if (val == 0x0) {
+			// Writer Fabric Enable
+			result = sysfs_write_u64_int(sysfs_path, 1);
```

Comment:
sysfs_write_u64_int is a new function?  How is it different than sysfs_write_u64?